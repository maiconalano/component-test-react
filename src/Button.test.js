import React from "react";
import Button from "./Button";
import { render, fireEvent, cleanup } from "@testing-library/react";

afterEach(cleanup);

const defaultProps = {
  onClick: jest.fn(),
  text: "Enviar",
};

test("Validar botão snapshot", () => {
  const { container } = render(<Button {...defaultProps} />);
  expect(container.firstChild).toMatchSnapshot();
});

test("Validar se o botão renderiza o texto padrão", () => {
  const { queryByText } = render(<Button {...defaultProps} />);
  expect(queryByText("Enviar")).toBeTruthy();
});

test("Validar se o botão renderiza o texto modificado", () => {
  const { queryByText, rerender } = render(<Button {...defaultProps} />);
  rerender(<Button {...defaultProps} text="Enviar projeto" />);
  expect(queryByText("Enviar projeto")).toBeTruthy();
});

test("Validar chamada da função on click", () => {
  const onClick = jest.fn();
  const { getByText } = render(<Button {...defaultProps} onClick={onClick} />);
  fireEvent.click(getByText(defaultProps.text));
  expect(onClick).toHaveBeenCalled();
});
