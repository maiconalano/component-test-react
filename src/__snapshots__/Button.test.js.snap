// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validar bot√£o snapshot 1`] = `
<button>
  Enviar
</button>
`;
